src/content/en/2020/http2.md

```diff
@@ -4,20 +4,534 @@ part_number: IV
+
+A [test suite](https://github.com/andydavies/http2-prioritization-issues) and a list of how various CDN/Cloud Hosting services perform was created. The bad news is that only 9 of the 36 prioritize correctly. Figure 22.16 shows that for sites using a CDN, around 31.7% do not prioritize correctly. This is up from 26.82% last year, mainly due to the increase in Google CDN usage. Rather than relying on the browser-sent priorities, there are some servers that implement a [server side prioritization](https://blog.cloudflare.com/better-http-2-prioritization-for-a-faster-web/) scheme instead, improving upon the browser’s hints with additional logic.
+
+<figure markdown>
+| CDN               | Prioritize correctly | Desktop | Mobile |
+|-------------------|---------|--------|--------|
+| Not using CDN     | Unknown | 59.47% | 60.85% |
+| Cloudflare        | Pass    | 22.03% | 21.32% |
+| Google            | Fail    | 8.26%  | 8.94%  |
+| Amazon CloudFront | Fail    | 2.64%  | 2.27%  |
+| Fastly            | Pass    | 2.34%  | 1.78%  |
+| Akamai            | Pass    | 1.31%  | 1.19%  |
+| Automattic        | Pass    | 0.93%  | 1.05%  |
+| Sucuri Firewall   | Fail    | 0.77%  | 0.63%  |
+| Incapsula         | Fail    | 0.42%  | 0.34%  |
+| Netlify           | Fail    | 0.27%  | 0.20%  |
+
+<figcaption>{{ figure_link(caption="HTTP/2 prioritization support in common CDNs.", sheets_gid="1152953475", sql_file="percentage_of_h2_and_h3_sites_affected_by_cdn_prioritization.sql") }}</figcaption>
+</figure>
+
+For non-CDN usage, I expect the number of servers that correctly apply HTTP/2 prioritization to be considerably smaller. For example, [NodeJS’s HTTP/2 implementation does not even support prioritization](https://twitter.com/jasnell/status/1245410283582918657).
+
+{# TODO consider rewording to come across as less of a dig. Or at least removing the word "even" #}
+
+### Goodbye Server push?
+
+Server Push was one of the additional features of HTTP/2 that caused some confusion and  complexity to implement in practice. Push seeks to avoid waiting for a browser/client to download a HTML page, parse that page, and only then discover that it requires a stylesheet. Once the stylesheet has been loaded, the browser parses that, only to realise it also needs a font. All these round-trips take time.  With server push, in theory, the server can just send multiple responses at once, saving the extra round trips.
```

Comment:
```suggestion
Server Push was one of the additional features of HTTP/2 that caused some confusion and  complexity to implement in practice. Push seeks to avoid waiting for a browser/client to download a HTML page, parse that page, and only then discover that it requires additional resources (such as a stylesheet), which in turn have to be fetched and parsed to discover even more dependencies (such as fonts). All that work and round trips takes time.  With server push, in theory, the server can just send multiple responses at once, avoiding the extra round trips.
```

The example is good but it reads a bit too specifically, push can be (mis)used for a lot of resource types.